docs/tutorials/storage-bucket.md

```diff
@@ -0,0 +1,125 @@
+
+# Create the bucket with uniform access and no public exposure.
+gcloud storage buckets create "$BUCKET" \
+  --project "$PROJECT_ID" \
+  --location "$REGION" \
+  --uniform-bucket-level-access \
+  --default-storage-class=STANDARD  # add --autoclass to enable automated tiering when you can tolerate slower cold reads
+
+# Grant yourself (or a service account) Storage Admin if needed.
+gcloud storage buckets add-iam-policy-binding "$BUCKET" \
+  --member="user:you@example.com" \
+  --role="roles/storage.objectAdmin"
+```
+
+Uniform bucket-level access ensures IAM policies apply consistently; keep the bucket private unless you intentionally publish checkpoints.
+
+## Step 3: Disable Soft Delete
+
+
+!!! warning
+   Disabling soft delete is critical to avoid runaway storage costs. Marin creates many large, short-lived files that should be deleted immediately.
+   Of course, disabling soft delete means you cannot recover deleted files, so consider implementing lifecycle rules or replication for backups if needed.
+
+GCS enables soft delete by default on new buckets. That feature retains deleted objects for at least seven days, which quickly explodes storage usage for Marin/Levanter workloads because training jobs constantly create and remove multi-gigabyte checkpoints and compilation caches. Disable soft delete immediately after creating the bucket:
+
+```bash
+# Permanently disable soft delete for this bucket.
+gcloud storage buckets update "$BUCKET" --clear-soft-delete
+
+# Optional: verify that the policy is cleared.
+gcloud storage buckets describe "$BUCKET" \
+  --format="value(soft_delete_policy)"
+```
+
+Clearing the policy ensures that once a training job deletes temporary files they disappear immediately, preventing runaway storage bills. You can still enable backups via lifecycle rules or replication if you need recovery.
+
+## Step 4: Configure Lifecycle Rules (Optional but Helpful)
+
+Large experimentation buckets benefit from automatic cleanup. For example, delete files under `tmp/` after seven days and move old checkpoints to colder storage:
+
+```json
+{
+  "rule": [
+    {
+      "action": {"type": "Delete"},
+      "condition": {"age": 7, "matchesPrefix": ["tmp/"]}
+    },
+    {
+      "action": {"type": "SetStorageClass", "storageClass": "ARCHIVE"},
+      "condition": {"age": 30, "matchesPrefix": ["checkpoints/"]}
+    }
+  ]
+}
+```
+
+If you chose autoclass storage, you can skip the second rule since Google handles tiering automatically.
+
+Save this as `lifecycle.json` and apply it:
+
+```bash
+gsutil lifecycle set lifecycle.json "$BUCKET"
+```
+
+Adjust prefixes to match how your experiments organize outputs.
+
+## Step 5: Wire It Into Marin / Levanter
+
+Set the bucket as your default prefix whenever you run tutorials:
+
+```bash
+export MARIN_PREFIX=$BUCKET
+export WANDB_PROJECT=marin
+export WANDB_ENTITY=your-entity
+```
+
+For Levanter configs, point the checkpointer to the same bucket:
+
+```yaml
+trainer:
```

Comment:
Just so I understand  this if I'm running a levanter model manually? Normally I'd construct the run via a Marin experiment file and programatically generate this?